external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriter.scala

```diff
@@ -101,4 +86,17 @@ private[kafka010] object KafkaWriter extends Logging {
         finallyBlock = writeTask.close())
     }
   }
+
+  private def validateAttribute(
```

Comment:
So this looks like exactly the same thing you have in `KafkaWriterTask`. You could even use the same one-method approach there, as far as I can see, instead of calling `expression` + `assertDataType`.

If the goal is to deduplicate code, then here's another one you can deduplicate.

Reply:
Nice suggestion. I'll need to see where it's the good place to put. Thanks!